sled-agent/src/instance.rs

```diff
@@ -385,6 +433,34 @@ impl InstanceRunner {
                                 warn!(self.log, "InstanceRunner failed to send to InstanceMonitorRunner");
                             }
                         },
+                         Some(InstanceMonitorRequest { update: Error(code), tx }) => {
+                            // TODO(eliza): this is kinda jank but Propolis
+                            // currently could return the same error when the
+                            // instance has not been ensured and when it's still
+                            // coming up, which is sad. I'm not sure whether
+                            // that happens IRL or not --- investigate this.
+                            let reaction = if has_ever_seen_a_good_state {
```

Comment:
This extra junk is because I noticed that Propolis' instance-state-monitor endpoint will presently return the `NoInstance` error on both `VmError::NotCreated` _and_ `VmError::WaitingToInitialize`:

https://github.com/oxidecomputer/propolis/blob/11371b0f3743f8df5b047dc0edc2699f4bdf3927/bin/propolis-server/src/lib/server.rs#L397-L399

This gave me some anxiety about sled-agent potentially deciding that a Propolis has "died" while it's actually still waiting to be born, although I'm not sure whether we would actually see the `WaitingToInitialize` error there in practice or not. I think the proper thing to do would be to change that behavior in Propolis, but I wanted to get something working in sled-agent regardless.

Reply:
after actually looking in Propolis, it looks like the `Vm::state_watcher()` function will never actually return `VmError::WaitingToInitialize`; it will only return `VmError::NotCreated`: 
https://github.com/oxidecomputer/propolis/blob/11371b0f3743f8df5b047dc0edc2699f4bdf3927/bin/propolis-server/src/lib/vm/mod.rs#L353-L364

So, hitting the state-monitor endpoint should just return a `Starting` VMM state in that case, which is good. I still don't like that the method _can_ return any of the error types, though, even though it doesn't currently --- I'll go try to clean that up on the Propolis side.

Reply:
Okay, https://github.com/oxidecomputer/propolis/pull/771 will make the propolis side way less scary. But I think this code can also just be removed as propolis' current behavior is also correct.